### PR TITLE
Add `voided_at` and `voided_by_user` to `ParticipantDeclaration`

### PR DESCRIPTION
**Pull request by ethax-ross**
[Jira-4084](https://dfedigital.atlassian.net.mcas.ms/browse/CPDLP-4084)

### Context

We want to maintain a record of who voided a declaration when it happens through the user interface.

### Changes proposed in this pull request

- Add `voided_at` and `voided_by_user` to `ParticipantDeclaration` 

These will be `nil` when a lead provider has voided the declaration via the API.

### Guidance to review

The `voided_at` column appears to have been part of the model years ago, which is why it was later ignored - see #2320 - reversing that change here as we re-introduce it.

### CODE REVIEW

File: spec/models/participant_declaration_spec.rb [hunk 1]
```diff
@@ -22,6 +23,18 @@
     it { is_expected.to validate_presence_of(:declaration_date) }
     it { is_expected.to validate_presence_of(:declaration_type) }
     it { is_expected.to validate_presence_of(:cohort) }
+
+    context "when voided_at is present" do
```
Comment (cwrw) [inline spec/models/participant_declaration_spec.rb:27]: is it worth adding a test that if both are `nil` that the declaration is valid?

Comment (ethax-ross) [reply spec/models/participant_declaration_spec.rb:27]: I'll add explicit tests to ensure the fields are not validated when nil 👍

File: app/models/participant_declaration.rb [hunk 2]
```diff
@@ -45,6 +46,8 @@ class ParticipantDeclaration < ApplicationRecord
   delegate :fundable?, to: :participant_profile, allow_nil: true
 
   validates :course_identifier, :user, :cpd_lead_provider, :declaration_date, :declaration_type, :cohort, presence: true
+  validates :voided_by_user, presence: true, if: :voided_at
+  validates :voided_at, presence: true, if: :voided_by_user
```
Comment (leandroalemao) [inline app/models/participant_declaration.rb:50]: Doesn't it create a dead lock condition? Not sure.. as it validates the other when the first present and vice-versa?

Comment (ethax-ross) [reply app/models/participant_declaration.rb:50]: Not sure I follow, I can't see how it could deadlock? It'll process the validation rules in order

**Review by mooktakim — APPROVED** [review]
